### ISSUE TITLE
[opencv4] Build the opencv4 with the MinGW compiler.

### ISSUE THREAD

**Issue by jeff665547** [microsoft/vcpkg#25775]
****

The current opencv4 ports cannot be built successfully with the mingw triplet on the windows platform.
The problem is related to the issue: https://github.com/microsoft/vcpkg/issues/23165 which is not solved actually.

I want to add the following code into the portfile.cmake file under the opencv4 port directory.
```
if(NOT MSVC)
  message(WARNING "The compiler is not Microsoft Visual Studio C/C++")
  message(WARNING "This compiler is not supported Microsoft Media Foundation (MSMF)")
  set(WITH_MSMF OFF)
endif()
```
The above code snippet will be added below the following section in the portfile.cmake:
```
set(WITH_MSMF ON)
if(NOT VCPKG_TARGET_IS_WINDOWS OR VCPKG_TARGET_IS_UWP)
  set(WITH_MSMF OFF)
endif()
```
The solution is based on the [stackoverflow](https://stackoverflow.com/questions/50333886/opencv-installation-on-windows). 

Could someone tell me is this proposed feature valid under all scenarios? Thanks!

**Comment by longnguyen2004** [comment]
The `if(NOT MSVC)` is wrong, since CMake is running in script mode, therefore it has no notion of a compiler.
Instead, you can change the existing condition, from `if(NOT VCPKG_TARGET_IS_WINDOWS OR VCPKG_TARGET_IS_UWP)` to `if(NOT VCPKG_TARGET_IS_WINDOWS OR VCPKG_TARGET_IS_UWP OR VCPKG_TARGET_IS_MINGW)`
I'll probably go back and investigate why it failed, since mingw-w64 is supposed to provide Media Foundation headers.

**Comment by dg0yt** [comment]
This issue was fixed with #25431.

**Comment by LilyWangLL** [comment]
Thanks for posting this issue. This issue has been fixed, so I am closing this issue for now.
